controllers/snapshot/snapshot_adapter_test.go

```diff
@@ -969,6 +970,136 @@ var _ = Describe("Snapshot Adapter", Ordered, func() {
 		})
 	})
 
+	When("Snapshot is Invalid by way of oversized name", func() {
+
+		BeforeEach(func() {
+			hasInvalidSnapshot = &applicationapiv1alpha1.Snapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "this-name-is-too-long-it-has-64-characters-and-we-allow-max-63ch",
+					Namespace: "default",
+					Labels: map[string]string{
+						gitops.SnapshotTypeLabel:              "component",
+						gitops.SnapshotComponentLabel:         "component-sample",
+						"build.appstudio.redhat.com/pipeline": "enterprise-contract",
+						gitops.PipelineAsCodeEventTypeLabel:   "push",
+					},
+					Annotations: map[string]string{
+						gitops.PipelineAsCodeInstallationIDAnnotation:   "123",
+						"build.appstudio.redhat.com/commit_sha":         "6c65b2fcaea3e1a0a92476c8b5dc89e92a85f025",
+						"appstudio.redhat.com/updateComponentOnSuccess": "false",
+					},
+				},
+				Spec: applicationapiv1alpha1.SnapshotSpec{
+					Application: hasApp.Name,
+					Components: []applicationapiv1alpha1.SnapshotComponent{
+						{
+							Name:           "component-sample",
+							ContainerImage: sample_image,
+							Source: applicationapiv1alpha1.ComponentSource{
+								ComponentSourceUnion: applicationapiv1alpha1.ComponentSourceUnion{
+									GitSource: &applicationapiv1alpha1.GitSource{
+										Revision: sample_revision,
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, hasInvalidSnapshot)).Should(Succeed())
+
+			Eventually(func() error {
+				err := k8sClient.Get(ctx, types.NamespacedName{
+					Name:      hasInvalidSnapshot.Name,
+					Namespace: "default",
+				}, hasInvalidSnapshot)
+				return err
+			}, time.Second*10).ShouldNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			err := k8sClient.Delete(ctx, hasInvalidSnapshot)
+			Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+		})
+
+		It("will stop reconciliation", func() {
+			var buf bytes.Buffer
+
+			integrationTestScenarioForInvalidSnapshot := &v1beta1.IntegrationTestScenario{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "invald-snapshot-its",
+					Namespace: "default",
+
+					Labels: map[string]string{
+						"test.appstudio.openshift.io/optional": "false",
+					},
+				},
+				Spec: v1beta1.IntegrationTestScenarioSpec{
+					Application: "application-sample",
+					ResolverRef: v1beta1.ResolverRef{
+						Resolver: "git",
+						Params: []v1beta1.ResolverParameter{
+							{
+								Name:  "url",
+								Value: "https://github.com/redhat-appstudio/integration-examples.git",
+							},
+							{
+								Name:  "revision",
+								Value: "main",
+							},
+							{
+								Name:  "pathInRepo",
+								Value: "pipelineruns/integration_pipelinerun_pass.yaml",
+							},
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, integrationTestScenarioForInvalidSnapshot)).Should(Succeed())
```

Comment:
We miss cleanup of this CR at the end of test

Reply:
Why it's not part of test provisioning in BeforeEach ?

Reply:
Thanks for the good catch, I have refactored it so it is included in the BeforeEach and cleanup for that test [here](https://github.com/redhat-appstudio/integration-service/pull/509/files#diff-6b881c00b61dec8580ed96e027cdd8c99d98abe475aacdcf351056fe41852732R1012) and [here](https://github.com/redhat-appstudio/integration-service/pull/509/files#diff-6b881c00b61dec8580ed96e027cdd8c99d98abe475aacdcf351056fe41852732R1056)